### PR TITLE
HAWQ-967. Extend Projection info to include filter attributes.

### PR DESCRIPTION
**Pull request by sansanichfb**

### CODE REVIEW

**Comment by GodenYao** [comment]
+1

File: src/backend/access/external/pxffilters.c [hunk 1]
```diff
@@ -451,6 +452,51 @@ opexpr_to_pxffilter(OpExpr *expr, PxfFilterDesc *filter)
 	return false;
 }
 
+static List*
+append_attr_from_var(Var* var, List* attrs)
+{
+	AttrNumber varattno = var->varattno;
+	/* system attr not supported */
+	if (varattno > InvalidAttrNumber)
```
Comment (kavinderd) [inline src/backend/access/external/pxffilters.c]: So the `varattno` is only valid when greater than `InvalidAttrNumber`?

**Comment by kavinderd** [comment]
Nice work! LGTM

File: src/backend/access/external/pxffilters.c (hunk 1 above)
Comment (sansanichfb) [inline src/backend/access/external/pxffilters.c]: @kavinderd, yes, physical attributes should have positive value, some system attributes have -1, and we are not projecting them.

**Pull request closed without merging**